prowler/providers/aws/services/opensearch/opensearch_service_domains_not_publicly_accessible/opensearch_service_domains_not_publicly_accessible.py

```diff
@@ -17,7 +17,10 @@ def execute(self):
             report.status_extended = (
                 f"Opensearch domain {domain.name} does not allow anonymous access."
             )
-            if domain.access_policy:
+
+            if domain.vpc_id:
+                report.status_extended = f"Opensearch domain {domain.name} is in a VPC, then it is not publicly accessible."
```

Comment:
Please, check if the domain is in a VPC public subnet or in a private one.

Reply:
I don't think it's sufficient to check whether the subnet is public or private. Even if a subnet is public, the domain won't necessarily be public without additional configurations, such as AWS PrivateLink, a VPN, or a proxy. For more details, you can refer to this discussion: [Access to AWS OpenSearch in a VPC](https://repost.aws/questions/QUMsQz3kFkSUKIIXVRmxSh2A/access-to-aws-opensearch-in-a-vpc).

Simply checking the subnet type doesn't definitively determine whether the domain should pass or fail. To avoid false positives, we need to perform a more comprehensive check.

Reply:
So if I have an OpenSearch in a VPC inside a public subnet exposed to the public, it is not publicly accessible?

Reply:
Yes, I checked with real infrastructure and I was not able to connect even the subnet was public. In this [docs](https://docs.aws.amazon.com/opensearch-service/latest/developerguide/vpc.html#vpc-architecture) there are some ways to access but you have to configure some things, by default there is no access. I think with the configurations like VPN or proxy the check still being a PASS because this methods are not bad practices however they are "exposing" the domain to Internet.

Reply:
Awesome, thanks for the testing!